server/bleep/src/indexes/file.rs

```diff
@@ -246,28 +244,31 @@ impl Indexable for File {
         let start = std::time::Instant::now();
 
         use rayon::prelude::*;
-        walker.par_iter().for_each(|file_disk_path| {
-            let workload = Workload {
-                file_disk_path: file_disk_path.clone(),
-                repo_disk_path: &repo.disk_path,
-                repo_ref: reporef.to_string(),
-                repo_name: &repo_name,
-                cache: &file_cache,
-                repo_info,
-            };
-
-            debug!(?file_disk_path, "queueing file");
-            if let Err(err) = self.worker(workload, writer) {
-                warn!(%err, ?file_disk_path, "indexing failed; skipping");
-            }
-        });
+        walker
+            .into_par_iter()
+            .chain(Some(repo.disk_path.clone()))
```

Comment:
when walking some path `p`, does the walker not add `p` itself to the results?

Reply:
Good point, I misunderstood the behaviour. Fixed in [a4f1e18](https://github.com/BloopAI/bloop/commit/a4f1e1838538bd1675d81a51c85f9a6e846dc54b)